src/server/routes/challenges/delete-challenge.ts

```diff
@@ -0,0 +1,21 @@
+import { Request, IReply } from 'hapi'
+import { ChallengeModel } from '../../models'
+import * as Boom from '../../boom'
+
+export default async function handler(req: Request, reply: IReply) {
+  const challengeId = req.params.challengeId
+
+  if (challengeId === undefined || typeof challengeId !== 'string' || challengeId.length === 0) {
+    reply(Boom.badRequest())
+    return
+  }
+
+  const challenge = await ChallengeModel.findOne({ challengeid: challengeId }).exec()
```

Comment:
It's there so the route can return 404. Got a better suggestion?

[`Query#remove([criteria], [callback])`](http://mongoosejs.com/docs/api.html#query_Query-remove)

Seems it returns `writeOpResult` which _may_ have something I can look for? These typings are outdated though.